### PR TITLE
Support string parameters in python operators

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: ocos/pyfunc/pyfunc.cc [hunk 1]
```diff
@@ -196,11 +196,11 @@ struct PyCustomOpDefImpl : public PyCustomOpDef {
     return obj;
   }
 
-  static py::object InvokePyFunction(uint64_t id, const py::object& feed) {
-    return (*op_invoker)(id, feed);
+  static py::object InvokePyFunction(uint64_t id, const py::object& feed, const py::object& atts) {
+    return (*op_invoker)(id, feed, atts);
```
Comment (wenbingl) [inline ocos/pyfunc/pyfunc.cc:200]: attr? since feed is also plural

File: ocos/pyfunc/pykernel.h [hunk 2]
```diff
@@ -36,10 +37,13 @@ struct PyCustomOpDef {
 };
 
 struct PyCustomOpKernel {
-  PyCustomOpKernel(OrtApi api, uint64_t id)
+  PyCustomOpKernel(OrtApi api, const OrtKernelInfo* info, uint64_t id, const std::vector<std::string>& attribute_names)
       : api_(api),
         ort_(api_),
         obj_id_(id) {
+    for (std::vector<std::string>::const_iterator it = attribute_names.begin(); it != attribute_names.end(); ++it) {
+      attribute_values_[*it] = ort_.KernelInfoGetAttribute<std::string>(info, it->c_str());
```
Comment (wenbingl) [inline ocos/pyfunc/pykernel.h:45]: Error-handling if the attribute is missing in the model

File: test/test_pyops.py [hunk 3]
```diff
@@ -115,6 +134,17 @@ def negpos(x):
             pos[x < 0] = 0
             return neg, pos
 
+        @onnx_op(op_type="PyOpJoin",
+                 inputs=[PyCustomOpDef.dt_string],
+                 outputs=[PyCustomOpDef.dt_string],
+                 atts=['sep'])
```
Comment (wenbingl) [inline test/test_pyops.py:140]: attributes?
Can this fields be fetch from ONNX model?

File: ocos/pyfunc/pyfunc.cc (hunk 1 above)
Comment (xadupre) [reply ocos/pyfunc/pyfunc.cc:200]: done

**Review by wenbingl — APPROVED** [review]

**Pull request merged**